src/mono/wasm/runtime/driver.c

```diff
@@ -206,15 +205,15 @@ mono_wasm_add_assembly (const char *name, const unsigned char *data, unsigned in
 	return mono_has_pdb_checksum (data, size);
 }
 
-EMSCRIPTEN_KEEPALIVE int
+int
 mono_wasm_assembly_already_added (const char *assembly_name)
 {
 	if (assembly_count == 0)
 		return 0;
 
 	WasmAssembly *entry = assemblies;
 	while (entry != NULL) {
-		if (strcmp (entry->assembly.name, assembly_name) == 0)
+		if (strncmp (entry->assembly.name, assembly_name, strlen(entry->assembly.name - 4)) == 0)
```

Comment:
hmmm isn't this is potentially fragile if the assemblies are MyAssembly.dll and MyAssembly.Browser.dll?  Should at least check that the lengths match as well.